### DISCUSSION TITLE
package.xml is being constructed with files outside of the force-app directory

### DISCUSSION THREAD

**Discussion by sheidasf** [scolladon/sfdx-git-delta#918] [General]
Hello

in our repo we have a folder outside of force-app called Archive. When we deprecate a class or anything, we 'rename' and move it to the arhive.

i noticed that when package.xml is constructed the files that are being renamed are picked up and added to the package.xml and the '--output ' directory contains that file under the 'archive'.

so i have two quesitons:
1- how come that file was picked up to be included in package.xml when the file is moved outside of the force-app in the repo
2- how come salesforce is not complaining that i ahve an entry in package.xml and it is not found under force-app.

Thank you so much for your time. and if i am being confusing, i can clarify further.

**Comment by scolladon** [comment]
HIi @sheidasf !

Here my answers to your questions:

> how come that file was picked up to be included in package.xml when the file is moved outside of the force-app in the repo

The plugin pick any file changed / added / deleted matching metadata inside the repository, wherever they are. It does not look for the `packageDirectories` attribut of the `sfdx-project.json` file.
It could ! I can add a feature where it uses a file pass in parameter to scope the analysis.

In your situation I see 2 ways to deal with this situation.
You could either : 
- use the parameter `-s force-app` (`--source`) to ask the plugin to specifically look for changes inside this folder
- use the parameter `-i` (`--ignore`) to tell the plugin to ignore the `archive` folder. You can pass a file following gitignore structure.

> how come salesforce is not complaining that i ahve an entry in package.xml and it is not found under force-app.

Maybe you ignore warnings ?

**Reply by sheidasf** [reply]
Sebastian, you are awesome. thank you so much. the --source parameter solved the problem and i am no longer seeing those items in the pakcage.xml
